corehq/apps/export/models/deid_export.py

```diff
@@ -0,0 +1,42 @@
+import hashlib
+import uuid
+
+from django.db import models
+
+from corehq.const import ONE_DAY
+from corehq.util.quickcache import quickcache
+
+
+class DeIdMapping(models.Model):
+    domain = models.TextField(max_length=255)
+    hashed_value = models.TextField(max_length=32)
+    deid = models.UUIDField(default=uuid.uuid4)
+
+    class Meta:
+        indexes = [
+            models.Index(fields=['domain', 'hashed_value']),
+        ]
+
+    @classmethod
+    def get_deid(cls, value, doc, domain=None):
+        if doc is not None:
+            # use domain from the couch doc if one was passed in
+            domain = doc['domain']
+
+        return cls._get_deid(value, domain)
+
+    @classmethod
+    @quickcache(['value', 'domain'], timeout=90 * ONE_DAY)
```

Comment:
90 days is a long time. If the code behind this cache changes, it will not take effect until months later (for cached values). Can you explain more about why such a long cache timeout was chosen?

Reply:
I hadn't considered code changes effectively being delayed by cache, which is a good point, but the length of time is because these values are essentially expected to stay the same forever once stored. The reason for caching here generally is so we're not making excessive DB calls for every repeated "sensitive ID" value in an export.

Maybe there's a shorter timeout here that would be more convenient for consideration of working with this code in the future, but still mitigate any performance impact on exports. Maybe 1 month? 1 week?

Reply:
I think even one day or even one hour would be fine if the intent is to reduce database load.